pkg/diagnostic/k8s/check.go

```diff
@@ -22,20 +22,20 @@ import (
 
 const DiagnosticK8sVersion = config.DiagnosticK8sVersion
 
-type checker struct {
+type Checker struct {
```

Comment:
Allowing `Checker`  to be exported

Reply:
Actually you don't want to break this encapsulation. Notice the "NewProvider" passed back and interface with well known functions. 

I would recommend making a new interface is needed; then cast the returned pointer to that interface.


That said, I would also recommend moving `GetConfig` to a more generic method in the k8s package you just created.